privacyidea/lib/token.py

```diff
@@ -181,6 +185,10 @@ def _create_token_query(tokentype=None, realm=None, assigned=None, user=None,
             # exact match
             sql_query = sql_query.filter(func.lower(Token.tokentype) == tokentype.lower())
 
+    if token_type_list is not None and len(token_type_list) > 0:
```

Comment:
It first filters for the `token_type` and based on the found token types it filters for all types that are in the `token_types_list`. Hence, if `token_type` is also in the `token_types_list`, only tokens of `token_type` will be returned. If `token_type` is not in the list, no tokens are returned, because no token matches the search query. 
This behavior is required to filter the tokens for all supported types of a container and allow additionally the user to filter this list by passing a wildcard in the `token_type`parameter.